Dockerfile.dev

```diff
@@ -0,0 +1,22 @@
+FROM golang:1.17-alpine
+ENV GO111MODULE=on
+ENV CGO_ENABLED=0
+ENV GOOS=linux
+
+RUN apk add --no-cache make git bash
+
+WORKDIR /go/src/github.com/netlify/gotrue
```

Comment:
Thanks for putting this together! This will be a great help towards having a easy to set up dev environment. 

As a whole, this looks good to me. One thing to note is that we may unlink from Netlify's fork as per #444 . As such, we will probably need to change all the namespaces from `github.com/netlify/gotrue` to `github.com/supabase/gotrue`

Reply:
Just checked, prev comment probably isn't relevant for now. Please ignore -- feel free to go ahead with documentation :)

Reply:
:thumbsup: Sounds good. That was a very quick turn around :smiley: